broadinstitute/leonardo
# GAWB-2000: Implement PUT endpoint for creating clusters

This is the code that will let us create dataproc clusters.
The path for this endpoint is: PUT /clusters/<google_project>/<cluster_name>
The payload of this endpoint contains:
* a bucket path (String) where all the notebooks will be saved - currently does nothing)
* the service account (String) to assign to the cluster (this will the user's SA)
* labels (Map[String, String) to put on the clusters

In order to test the endpoint locally:
* Clone leo and checkout the `gawb2000_as` branch
* Get your configs in order - checkout the `leonardo-configs` branch of firecloud-develop
* Run `APP_NAME=leonardo ENV=dev INPUT_DIR=./ OUTPUT_DIR=/yourpath/leonardo/config ./configure.rb`
* Run `sbt -Dconfig.file=config/leonardo.conf  run` in leo
* Go to `localhost:8080` to get to the swagger page
* When calling the endpoint, you can leave the bucket path and label fields blank in the payload, but you must have a service account. You can use the rawls SA or the leo SA here
* The leo service account MUST have Owner permissions in your google project for any of this to work

  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
